crates/wit-parser/src/resolve.rs

```diff
@@ -1329,9 +1329,22 @@ impl Resolve {
         }
     }
 
-    fn include_stability(&self, stability: &Stability) -> bool {
+    fn include_stability(&self, stability: &Stability, pkg_id: Option<&PackageId>) -> bool {
         match stability {
-            Stability::Stable { .. } | Stability::Unknown => true,
+            Stability::Unknown => true,
+            Stability::Stable { since, feature } => {
+                if pkg_id
+                    .and_then(|id| self.packages.get(*id))
+                    .map(|p| &p.name.version)
+                    .is_some_and(|v| v.as_ref().is_some_and(|v| since > v))
+                {
+                    panic!("referring to an unreleased (future) package version in a feature gate is not allowed");
```

Comment:
This'll want to use `bail!` instead to return an error to get bubbled up eventually to the original API call

Reply:
Thanks for pointing this out -- saw some other `panic`s so naively thought it was preferred! Much prefer to `bail!`.

- [ ] remove `panic` and return `Result` from stability fn